EIPS/eip-4950.md

```diff
@@ -0,0 +1,87 @@
+---
+eip: 4950
+title: Entangled Tokens
+description: ERC-721 extension with two tokens minted that are tied together
+author: Victor Munoz (@victormunoz)
+discussions-to: https://ethereum-magicians.org/t/entangled-tokens/8702
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-03-28
+requires: 721
+---
+
+## Abstract
+This EIP standardizes an interface for creating entangled pairs of tokens. An entangled pair of tokens give his owners the right to operate with this common wallet.
+
+## Motivation
+The motivation is to provide an easy way to share a wallet through NFTs. So that the act of buying an NFT (in a marketplace) gives the buyer the privilege to have access to a given wallet. This wallet could have budget in many tokens, or even be the propietary of other NFTs.
+
+An use case is to keep contact between an artist and an buyer of its NFTs. If an artist T has created a digital piece of art P with an NFT, then T creates 2 entangled tokens A and B so that he keeps A and transfer B to P. By construction of entangled tokens, only one transfer is possible for them, thus the artist proofs he’s been the creator of P by sending a transaction to A that is visible from B. Otherwise, the owner of P might check the authenticity of the artist by sending a transaction to B so that the artist might proof by showing the outcome out of A.
+
+An extension of this first use case is when one user U and its creation in the form of an NFT T with wallet then the user U creates and transfers an entangled token A to T, and U keeps the entangled token B in the U’s wallet.  
+
+These applications of entangled tokens are envisaged to be useful for 
+1.	NFT authorship / art creation
+2.	Distribution of royalties by the creator.
+3.	Authenticity of a work of art: creation limited to the author (e.g. only 1000 copies if there are 1000 1000 entangled tokens in that NFT).
+4.	Usowners
+5.	Reformulation of property rights: the one who owns the property receives it without having to follow in the footsteps of the owners.
+6.	Identity: Only those credentials that have an entangled token with you are related to you.
+7.	Vreservers (value-reservers).
+
+
+## Specification
+An entangled token is an NFT (ERC-721) with a modification such that it mints exactly two tokens at the contract deployment, and the token owners have access to the contract wallet.
+
+
+Also, a `tokenTransfer` function is to be be added in order to allow the token owners to transact with the ERC20 tokens owned by the contract/NFT itself.
+
+## Rationale
+We decide to extend ERC721 (ERC1155 could be also possible) because the main purpose of this is to be compatible with current marketplaces platforms. This entangled NFTs will be listed in a marketplace, and the user who buys it will have then the possibility to transact with the wallet properties (fungible and non fungible tokens).

```

Comment:
```suggestion
We decide to extend [EIP-721](./eip-721.md) ([EIP-1155](./eip-1155.md) could be also possible) because the main purpose of this is to be compatible with current marketplaces platforms. This entangled NFTs will be listed in a marketplace, and the user who buys it will have then the possibility to transact with the wallet properties (fungible and non fungible tokens).

```